docs/content/posts/soup.md

```diff
@@ -0,0 +1,119 @@
+Even if you are using LLMs to parse and explain the traces for you, this can make it focus on what you really want to understand instead of drowning in message-handling mechanics.
+
+## TODO: ADD BAR PLOT HERE FOR THE EXAMPLE ABOVE
+
+## The Need for Speed
+The message soup version finds witnesses **at least 3x faster** than the bookkeeping approach, and that's in the worst case. For more complex scenarios, the message-per-message version would often time out or run out of memory entirely while the message soup version sailed through. This speed matters because faster executions mean faster debugging cycles and more extensive coverage of the state space. When you can explore more scenarios in the same amount of time, you're establishing more trust in your system. More speed, more trust, more sleep.
+
+## TODO: ADD REST OF BAR PLOTS HERE
+
+## When the Soup Gets Spicy: Byzantine Faults
+A powerful trick to model byzantine behavior is to simply pre-populate the message soup with all messages that byzantine nodes could ever send without modeling complex byzantine node logic. (We have used this in our earlier [blog post on Alpenglow](https://quint-lang.org/posts/alpenglow), where we wanted to analyze the issue of having too many faults.)
+The message soup approach handles the load that comes with this technique more elegantly than explicit bookkeeping: because our system only makes transitions on enabled steps, having a huge number of possibilities for byzantine messages doesn't cause combinatorial explosion. In the bookkeeping approach, every byzantine message creates a new state transition that the system has to explore, leading to exponential blowup. 
+The plot below shows resilience to byzantine messages injection. We can see that the frequency of observing witnesses is unaffected for the message soup approach, but the bookkeeping version fails to maintain this resilience. In the single-message version, there's a massive gap between the Byzantine and honest scenarios. Byzantine conditions severely degrade performance while the message soup approach shows barely any difference.
+## TODO: ADD LINE PLOT HERE
+When you're dealing with adversarial networks, what doesn't kill your protocol makes it stronger. The message soup cuts through the byzantine noise and helps you focus on what really matters: is your design still safe when under attack?
+
+
+List of blog posts/resources for consensus in Quint:
+- [Solana Alpenglow](https://quint-lang.org/posts/alpenglow)
+- [Espresso Hotshot](https://informal.systems/blog/espresso-hotshot-epoch-changes-in-quint-2025)
+- [ZKSync ChonkyBFT](https://protocols-made-fun.com/consensus/matterlabs/quint/specification/modelchecking/2024/07/29/chonkybft.html)
+- [Commonware Minimmit](https://github.com/commonwarexyz/monorepo/tree/main/pipeline/minimmit/quint)
```

Comment:
```suggestion
- Quint blog post on [Solana Alpenglow](https://quint-lang.org/posts/alpenglow)
- Informal's blog post on [Espresso Hotshot](https://informal.systems/blog/espresso-hotshot-epoch-changes-in-quint-2025)
- Igor Konnov's blog post on [ZKSync ChonkyBFT](https://protocols-made-fun.com/consensus/matterlabs/quint/specification/modelchecking/2024/07/29/chonkybft.html)
- Commonware's Quint spec on [Minimmit](https://github.com/commonwarexyz/monorepo/tree/main/pipeline/minimmit/quint)
```
- The [MonadBFT spec](https://github.com/informalsystems/monad-spec) discussed in this post